pkg/proxy/iptables/proxier.go

```diff
@@ -975,6 +975,16 @@ func (proxier *Proxier) syncProxyRules() {
 	if err != nil {
 		klog.ErrorS(err, "Failed to get node ip address matching nodeport cidrs, services with nodeport may not work as intended", "CIDRs", proxier.nodePortAddresses)
 	}
+	// nodeAddresses may contain dual-stack zero-CIDRs if proxier.nodePortAddresses is empty.
+	// Ensure nodeAddresses only contains the addresses for this proxier's IP family.
+	isIPv6 := proxier.iptables.IsIPv6()
+	for addr := range nodeAddresses {
+		if utilproxy.IsZeroCIDR(addr) && isIPv6 == netutils.IsIPv6CIDRString(addr) {
+			// if any of the addresses is zero cidr of this IP family, non-zero IPs can be excluded.
+			nodeAddresses = sets.NewString(addr)
+			break
+		}
```

Comment:
```suggestion
		if isIPv6 != netutils.IsIPv6CIDRString(addr) {
			// filter all the addresses that doesn't match the proxy IP family
			nodeAddresses.Delete(addr)
		}
```

Reply:
scratch this comment https://github.com/kubernetes/kubernetes/pull/107413#discussion_r780471673